### DISCUSSION TITLE
what is this for 'http://204.13.164.118:80/tor/status-vote/current/consensus' ?

### DISCUSSION THREAD

**Discussion by xiaoguazh** [srixivas/PcapXray#61] [Q&A]
Running this program, it output below message, where this comes from? what it is used for?

DEBUG:stem:Unable to download descriptors from 'http://204.13.164.118:80/tor/status-vote/current/consensus' (2 retries remaining): Failed to download from http://204.13.164.118:80/tor/status-vote/current/consensus (URLError): A connection attempt failed because the connected party did not properly respond after a period of time, or established connection failed because connected host has failed to respond
DEBUG:stem:Unable to download descriptors from 'http://193.23.244.244:80/tor/status-vote/current/consensus' (1 retries remaining): Failed to download from http://193.23.244.244:80/tor/status-vote/current/consensus (URLError): A connection attempt failed because the connected party did not properly respond after a period of time, or established connection failed because connected host has failed to respond
DEBUG:stem:Unable to download descriptors from 'http://128.31.0.39:9131/tor/status-vote/current/consensus': Failed to download from http://128.31.0.39:9131/tor/status-vote/current/consensus (URLError): A connection attempt failed because the connected party did not properly respond after a period of time, or established connection failed because connected host has failed to respond
Unable to retrieve the consensus: Failed to download from http://128.31.0.39:9131/tor/status-vote/current/consensus (URLError): A connection attempt failed because the connected party did not properly respond after a period of time, or established connection failed because connected host has failed to respond
Starting Graph Plotting

**Comment by srixivas** [comment]
That would be a part of the Tor module determining possible traffic by fetching the consensus document that contains the status of the relays. That error most likely occurs when your machine is offline, other functions/modules should work fine. 
Check [here](https://metrics.torproject.org/glossary.html#consensus), [here](https://tor.stackexchange.com/questions/4939/how-can-i-get-consensus-data-from-directory-authority-servers-for-research-purpo) and this [blog](https://jordan-wright.com/blog/2015/05/14/how-tor-works-part-three-the-consensus/) for more. ([similar](https://svn.nmap.org/nmap/scripts/tor-consensus-checker.nse) with nmap).
